// ds-ui-block/app.js

var util = require('util'),
    Dashboard = require('./dashboard'),
    FS = require('fs');

var App = module.exports = function (name, params, manifest) {

    this.manifest = manifest;
    this.name = name;
    this.params = params;
};

util.inherits(App, Dashboard);

// raw package json
(function () {

    this.getPackageJSON = function (name, callback) {

        FS.readFile('package.json', 'utf8', function (err, data) {

            if (err)
                return callback(err);

            callback(null, JSON.parse(data));
        });
    };
}).call(App.prototype);

// data
(function () {

    this.data = function (res, e, context, callback) {

        var datapoint = context.datapoint;

        if (!datapoint) return callback("invalid data point");

        if (!this.manifest.data || !this.manifest.data.datapoint) return callback("invalid data point");

        var datapoint = this.manifest.data.datapoint;

        e.loader.load.call(this, datapoint, context.params, callback);
    };
}).call(App.prototype);


// adapter
(function () {

    this.adapter = function (e, context, options, callback) {

        if (typeof options === "function") {
            callback = options;
            options = null;
        }

        var opt = options || {};
        opt.context = context;
        opt.app = this.name;

        e.user(context.user).adapter(e, opt, callback);
    };
}).call(App.prototype);
