// ds-ui-block/dashboard.js

var chance = require('chance'),
    Block = require('./block'),
    util = require('util');

var Dashboard = module.exports = function (name, params, manifest) {

    Block.call(this, name, params, manifest);
};

util.inherits(Dashboard, Block);

// tilesets
(function () {

    this.tilesets = function (res, e, context, callback) {

        callback(null, []);
    };
}).call(Dashboard.prototype);

// filters
(function () {

    this.filters = function (res, e, context, callback) {

        callback(null, []);
    };
}).call(Dashboard.prototype);

// tiles
(function () {

    this.tiles = function (res, e, context, callback) {

        callback(null, []);
    };
}).call(Dashboard.prototype);

// tiles
(function () {

    this.widget = function (name, callback) {

        var Widget = null;

        try {

            Widget = require(['ds', 'widget', name].join('-'));

        } catch (e) {
            return callback('widget not found');
        };

        return callback(null, Widget);
    }
}).call(Dashboard.prototype);

// info
(function () {

    this.color = function (res, e, context, callback) {

        return this.manifest.color || chance.color();
    };

    this.category = function (res, e, context, callback) {

        return this.manifest.category || "All";
    };

    this.tags = function (res, e, context, callback) {

        return this.manifest.tags || [""];
    };
}).call(Dashboard.prototype);
