// ds-widget-grid/manifest.js

var uuid = require('node-uuid');

module.exports = {

    title: "DS -Grid",

    params: {

        "dataProcess": "url", // (url | data)

        //"target": "http://localhost:5555/me/data/position/find?_pred=fx.select(%22dt%22,%22fg%22,%22strat%22,%22cust%22,%22sid%22,%22sn%22,%22ana%22,%22mvb%22,%22dpnl%22)",
        "target": null,

        "data": null,

        //specify the list of columns in which grouping is to be done
        //"group": ["ana", "cust"],
        "group": [],

        //search tab is provided
        //"search": true,
        "search": false,

        "themes" : null,

        "aggregates" : null,
        "xDimension" : null,
        "yDimension" : null,

        "paging": true,


        // dynamic grid id
        "id": function (params) {
            return 'a' + uuid.v1()
        },

        "width": "100%",

        "height": "100%",

        "colModel": null,
        "colTotals": true,
        "rowTotals":true,

         //sorting
         "sortable" :false,  //do not want column header to perform sorting

         "sortParam" :null,  //default value in which sorting to be performed

         "multisort" : "false", //column on multiple columns true

         customFormatter : function(cellValue, opts, rowObject)
         {

         },

         onSelectRow : null,
         gridComplete: null,
         onCellSelect:null,
         afterInsertRow:null,
         ondblClickRow:null,
         onHeaderClick:null,
         onPaging:null,
         onRightClickRow:null,
         onSelectAll:null,
         loadComplete:null,


         footerrow: null,
         userdata:null
    },

    "data": null,

    guid: null

};
