// ds-widget-grid/examples/manifest_formatting.js
module.exports = {

    title: "DS -Grid",

    params: {

         "dataProcess" : "url",
         "target" : "/me/data/position/find?_pred=fx.select(%22dt%22,%22fg%22,%22strat%22,%22cust%22,%22sid%22,%22sn%22,%22ana%22,%22mvb%22,%22dpnl%22)",

         // custom formatting
         //Col Model defines al these properties   - By default date format MM/dd/yyyy  Currency: thousand seprator with decimal value none.

         colModel : [{
             "name": "dt",          //mandatory when defining specific properties for coloums
             "label": "AsOfDate",       // not specified then same as name
             "formatter":'date',
             "formatoptions": { "srcformat": "ISO8601Long", "newformat": "m/d/Y h:i A" },
             "frozen": "true"
         }]


         //  {
         //
         //     name: undefined,
         //     label: undefined,
         //     width: undefined,
         //     formatter: undefined, // defined formatters and custom formatter function
         //     formatoptions : undefined,
         //     align: undefined, //left, right, center textalign
         //     resizable: undefined, // true/false
         //     sortable: undefined, // true/false
         //     classes: undefined, // Classes to the column  eg 'class1 class2'
         //     frozen : undefined // column would be frozen
         //     resizable: undefined // column can be resized, default true
         //     sortable: undefined // column can be sorted, default true
         //     sortfunc : undefined // custom sort function
         // }


         /*
              formatter : {
                   integer : {thousandsSeparator: " ", defaultValue: '0'},
                   number : {decimalSeparator:".", thousandsSeparator: " ", decimalPlaces: 2, defaultValue: '0.00'},
                   currency : {decimalSeparator:".", thousandsSeparator: " ", decimalPlaces: 2, prefix: "", suffix:"", defaultValue: '0.00'},
                   date : {
                     dayNames: [
                       "Sun", "Mon", "Tue", "Wed", "Thr", "Fri", "Sat",
                       "Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"
                     ],
                     monthNames: [
                       "Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec",
                       "January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"
                     ],
                     AmPm : ["am","pm","AM","PM"],
                     S: function (j) {return j < 11 || j > 13 ? ['st', 'nd', 'rd', 'th'][Math.min((j - 1) % 10, 3)] : 'th'},
                     srcformat: 'Y-m-d',
                     newformat: 'd/m/Y',
                     parseRe: /[Tt\\\/:_;.,\t\s-]/,
                     masks : {
                       ISO8601Long:"Y-m-d H:i:s",
                       ISO8601Short:"Y-m-d",
                       ShortDate: "n/j/Y",
                       LongDate: "l, F d, Y",
                       FullDateTime: "l, F d, Y g:i:s A",
                       MonthDay: "F d",
                       ShortTime: "g:i A",
                       LongTime: "g:i:s A",
                       SortableDateTime: "Y-m-d\\TH:i:s",
                       UniversalSortableDateTime: "Y-m-d H:i:sO",
                       YearMonth: "F, Y"
                     },
                     reformatAfterEdit : false
                   },
                   baseLinkUrl: '',
                   showAction: '',
                   target: '',
                   checkbox : {disabled:true},
                   idName : 'id'
          */

         /*
               function currencyFmatter (cellvalue, options, rowObject)
               {
                  // do something here
                  return "$"+cellvalue;
               }
        */

     },



    "data" : null,

    guid: null,

};
